### ISSUE TITLE
Please support gcc version 13.2 and 14.0 in scons - was: gcc 13 compilation error: 'void* __builtin_memmove(void*, const void*, long unsigned int)' forming offset 8 is out of the bounds [0, 8] [-Werror=array-bounds=] ?

### ISSUE THREAD

**Issue by smoe** [RosettaCommons/rosetta#29]
I do not get what is wrong since 8 is in [0,8]. Any idea what I sould try? I came across something alike
```
#if (__GNUC__ == 13 && __GNUC_MINOR__ == 2)
#pragma GCC diagnostic ignored "-Warray-bounds"
#endif
```
but this may hide something I am not able to see. clang 16 does not have any issue with that particular code.

The following error is reported many times in series until the compilation stops:
```
In file included from /usr/include/c++/13/bits/stl_uninitialized.h:63,
                 from /usr/include/c++/13/memory:69,
                 from src/utility/pointer/std/owning_ptr.hh:20,
                 from src/utility/pointer/owning_ptr.hh:15,
                 from src/protocols/antibody/constraints/ParatopeEpitopeSiteConstraintMover.fwd.hh:18,
                 from src/protocols/antibody/constraints/ParatopeEpitopeSiteConstraintMover.hh:17,
                 from src/protocols/antibody/constraints/ParatopeEpitopeSiteConstraintMover.cc:14:
In static member function 'static _Up* std::__copy_move<_IsMove, true, std::random_access_iterator_tag>::__copy_m(_Tp*, _Tp*, _Up*) [with _Tp = long unsigned int; _Up = long unsigned int; bool _IsMove = false]',
    inlined from '_OI std::__copy_move_a2(_II, _II, _OI) [with bool _IsMove = false; _II = long unsigned int*; _OI = long unsigned int*]' at /usr/include/c++/13/bits/stl_algobase.h:506:30,
    inlined from '_OI std::__copy_move_a1(_II, _II, _OI) [with bool _IsMove = false; _II = long unsigned int*; _OI = long unsigned int*]' at /usr/include/c++/13/bits/stl_algobase.h:533:42,
    inlined from '_OI std::__copy_move_a(_II, _II, _OI) [with bool _IsMove = false; _II = long unsigned int*; _OI = long unsigned int*]' at /usr/include/c++/13/bits/stl_algobase.h:540:31,
    inlined from '_OI std::copy(_II, _II, _OI) [with _II = long unsigned int*; _OI = long unsigned int*]' at /usr/include/c++/13/bits/stl_algobase.h:633:7,
    inlined from 'std::vector<bool, _Alloc>::iterator std::vector<bool, _Alloc>::_M_copy_aligned(const_iterator, const_iterator, iterator) [with _Alloc = std::allocator<bool>]' at /usr/include/c++/13/bits/stl_bvector.h:1306:28,
    inlined from 'void std::vector<bool, _Alloc>::_M_fill_insert(iterator, size_type, bool) [with _Alloc = std::allocator<bool>]' at /usr/include/c++/13/bits/vector.tcc:879:34,
    inlined from 'std::vector<bool, _Alloc>::iterator std::vector<bool, _Alloc>::insert(const_iterator, size_type, const bool&) [with _Alloc = std::allocator<bool>]' at /usr/include/c++/13/bits/stl_bvector.h:1206:16,
    inlined from 'void std::vector<bool, _Alloc>::resize(size_type, bool) [with _Alloc = std::allocator<bool>]' at /usr/include/c++/13/bits/stl_bvector.h:1252:10,
    inlined from 'virtual void protocols::antibody::constraints::ParatopeEpitopeSiteConstraintMover::parse_my_tag(protocols::moves::Mover::TagCOP, basic::datacache::DataMap&)' at src/protocols/antibody/constraints/ParatopeEpitopeSiteConstraintMover.cc:155:24:
/usr/include/c++/13/bits/stl_algobase.h:437:30: error: 'void* __builtin_memmove(void*, const void*, long unsigned int)' forming offset 8 is out of the bounds [0, 8] [-Werror=array-bounds=]
  437 |             __builtin_memmove(__result, __first, sizeof(_Tp) * _Num);
      |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
cc1plus: all warnings being treated as errors
scons: *** [build/src/release/linux/5.15/64/x86/gcc/13/default/protocols/antibody/constraints/ParatopeEpitopeSiteConstraintMover.os] Error 1
scons: *** [build/src/release/linux/5.15/64/x86/gcc/13/default/protocols/antibody/constraints/ParatopeSiteConstraintMover.os] Error 1
scons: building terminated because of errors.
```
Please instruct me on the kind of patch I should prepare - if any.

**Comment by smoe** [comment]
Just observed the same for g++ version 14 and then eventually found the demotion of that error as a warning at https://github.com/RosettaCommons/rosetta/blob/2c36cbc7108d85646ca5b8ddc89c29ac1ccde88e/source/tools/build/basic.settings#L450 
which addresses g++ 13.1, not 13.2 and 14.0 that I tried.
So, this indeed overcomes this particular hurdle. What should I do?

**Comment by roccomoretti** [comment]
Unfortunately, our scons setup doesn't really support compiler version ranges at the moment. So if there's adjustments needed for particular compiler versions, you need to add multiple blocks to basic.settings, one for each minor version of the compiler you want to treat.

**Comment by lyskov** [comment]
@roccomoretti this also affects GCC-12 builds: https://b3.graylab.jhu.edu/test/823195

**Comment by roccomoretti** [comment]
I think this should be fixed now with PR #36 -- please reopen if that's not the case.
